### PR TITLE
docs: update toolchain + development sections + emoji

### LINKED ISSUE #354: add missing documentation
Review and update the README / inline documentation, addressing any missing parts and refreshing any outdated sections:
  * [x] description of the changes related to the toolchain, gc, build process and dependencies - @radkomih 
  * [x] an overview of a module's structure and its usage (module, storage, configuration, constants, dispatch calls, genesis builder) - @failfmi 
  * [x] information on the supported metadata versions and the generation process - @failfmi
  * [x] an explanation of the benchmarking methodology, along with details on utility functions (dispatch, hooks, overhead) - @radkomih 
  * [x] comprehensive documentation for each newly added module - @failfmi

Additionally, check the notes from phase 2 for any relevant information that should be included: https://www.notion.so/limechain/Phase-2-Notes-eece850141b0450a884f556d74bda3fb?pvs=4

### PR DESCRIPTION
**Pull request by radkomih**
**Detailed description**:

**Which issue(s) this PR fixes**:
Fixes #354 

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: docs/docs/development/test.md [hunk 1]
```diff
@@ -13,18 +15,20 @@ make test
...
-To aid the debugging process, there is a set of imported functions that can be called within the Runtime to log messages.
+To aid the debugging process, there is a set of functions provided by the logger instance that can be called within the Runtime to log messages.
 
 ```go
-func Critical(message string) // logs and aborts the execution
-func Warn(message string)
-func Info(message string)
-func Debug(message string)
-func Trace(message string)
+logger = log.NewLogger()
```
Comment (failfmi) [inline docs/docs/development/test.md]: `:=`

File: docs/docs/overview/toolchain.md [hunk 2]
```diff
@@ -4,6 +4,28 @@ title: Toolchain
...
 
-Since we use a [modified version of TinyGo](https://github.com/LimeChain/tinygo) for compiling Go to Wasm, some of Go's
-language capabilities cannot be applied due to the limited support in TinyGo which also affects some of the design
-decisions.
+# Toolchain 🛠️
+
+Since we use toolchain based on [modified version of TinyGo](https://github.com/LimeChain/tinygo) for compiling Go to Wasm, some of Go's
+language capabilities cannot be applied due to the limited support in TinyGo which also affects some of the design decisions.
+
+## 1. Tinygo modifications
```
Comment (failfmi) [inline docs/docs/overview/toolchain.md]: `TinyGo`

File: docs/docs/reference/resources.md [hunk 3]
```diff
@@ -4,11 +4,15 @@ title: Resources
...
+# Resources 📚
+ 
 ### Polkadot
 
-* [General](https://polkadot.network/)
+* [Official Website](https://polkadot.network/)
 * [Wiki](https://wiki.polkadot.network/)
 * [Specification](https://spec.polkadot.network/)
+* [Forum](https://forum.polkadot.network/)
+* [Stackexchange](https://substrate.stackexchange.com/)
```
Comment (failfmi) [inline docs/docs/reference/resources.md]: `Stack Exchange`

File: docs/docs/reference/resources.md [hunk 4]
```diff
@@ -18,6 +22,11 @@ permalink: /reference/resources
 
 * [Research](https://web3.foundation/research/)
 
+### Github
+
+* [Polkadot Fellows](https://github.com/polkadot-fellows)
+* [Paritytech](https://github.com/paritytech)
```
Comment (failfmi) [inline docs/docs/reference/resources.md]: Parity Technologies

File: docs/mkdocs.yml [hunk 5]
```diff
@@ -147,13 +147,14 @@ nav:
       - Toolchain: ./overview/toolchain.md
       - Modules: ./overview/modules.md
   - Development:
-      - Onboarding: ./development/onboarding.md
       - Install: ./development/install.md
       - Build: ./development/build.md
       - Inspect: ./development/inspect.md
-      - File structure: ./development/file-structure.md
-      - Testing: ./development/test.md
+      - Test: ./development/test.md
```
Comment (failfmi) [inline docs/mkdocs.yml]: Let's rename this back to `Testing`

Comment (failfmi) [reply docs/mkdocs.yml]: Also, I think testing and benchmark should be after the file structure.

File: docs/docs/development/benchmark.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+It exposes additional utility functions exported by the runtime, which allow the execution of benchmark tests in a Wasm environment.
+
+```bash
+"Benchmark_dispatch": [I32, I32] -> [I64]
+"Benchmark_hook": [I32, I32] -> [I64]
+```
+
+### 2. Write benchmarks 📝
```
Comment (failfmi) [inline docs/docs/development/benchmark.md]: There are two `2.`

File: docs/docs/development/benchmark.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+* Example benchmark test:
+[benchmark_timestamp_set_test.go](https://github.com/LimeChain/gosemble/blob/develop/runtime/benchmark_timestamp_set_test.go)
+
+* Example benchmark test with linear components:
+[benchmark_system_remark_test.go](https://github.com/LimeChain/gosemble/blob/develop/runtime/benchmark_timestamp_set_test.go)
+
+Extrinsic calls are executed through the `Benchmark_dispatch` runtime function.
+
+### 1.2. System hooks 🪝
```
Comment (failfmi) [inline docs/docs/development/benchmark.md]: Should these be `1.2.` and `1.3.`

File: docs/docs/reference/resources.md [hunk 8]
```diff
@@ -18,6 +22,11 @@ permalink: /reference/resources
 
 * [Research](https://web3.foundation/research/)
 
+### Github
+
+* [Polkadot Fellows](https://github.com/polkadot-fellows)
```
Comment (failfmi) [inline docs/docs/reference/resources.md]: Polkadot Fellowship

File: docs/docs/development/benchmark.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+It is important to note that benchmark tests should always assess the **worst-case** scenario. The general process of writing a benchmark test includes setting up an initial state, executing an operation, and asserting the final state, which encompasses both success and failure scenarios.
+
+### 2.1. Dispatch calls 📞
+
+* Example benchmark test:
+[benchmark_timestamp_set_test.go](https://github.com/LimeChain/gosemble/blob/develop/runtime/benchmark_timestamp_set_test.go)
+
+* Example benchmark test with linear components:
+[benchmark_system_remark_test.go](https://github.com/LimeChain/gosemble/blob/develop/runtime/benchmark_timestamp_set_test.go)
```
Comment (failfmi) [inline docs/docs/development/benchmark.md]: Link here is the same as the one above

File: docs/docs/development/benchmark.md
Comment (failfmi) [inline docs/docs/development/benchmark.md:1]: nit: rename to benchmarking

**Review by failfmi — DISMISSED** [review]

**Review by failfmi — APPROVED** [review]

File: docs/docs/development/onboarding.md [hunk 10]
```diff
@@ -234,9 +218,9 @@ Developing a framework for writing Polkadot runtimes in Go is not a straight for
 
 Most of the things are documented here [Gosemble Runtime Architecture](/overview/runtime-architecture/).
 
-## 4. Tasks
+## 4. Tasks 📝
```
Comment (tgntr) [inline docs/docs/development/onboarding.md:221]: nit: Guides/Tutorials or something?

File: docs/docs/development/benchmarking.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+## 4. Run benchmarks ▶️
+
+Run extrinsic and hook benchmarks.
+
+```bash
+make benchmark
+```
+```bash
+make benchmark steps=50 repeat=100
```
Comment (tgntr) [inline docs/docs/development/benchmarking.md:93]: steps and repeat flags are not present at this moment

File: README.md [hunk 12]
```diff
@@ -15,7 +15,7 @@ the [Official Documentation](https://limechain.github.io/gosemble/)
 - [Git](https://git-scm.com/downloads)
 - [Go 1.21](https://golang.org/doc/install)
 - [Docker](https://docs.docker.com/install/)
-- [Rust](https://docs.substrate.io/install/)
+- [Rust](https://docs.substrate.io/install/) (for building the Substrate node)
```
Comment (tgntr) [inline README.md:18]: nit: maybe just [Substrate](url)?

**Pull request merged**